### DISCUSSION TITLE
cmake时无报错，但是运行时单独输出“syntax error”而无其他信息

### DISCUSSION THREAD

**Discussion by RegiaYoung** [arcsysu/SYsU-lang#93] [Q&A]
![image](https://user-images.githubusercontent.com/68263210/230817725-1971ae7c-9e71-440c-a9ac-f69ace5344b9.png)

大概上面这种情况，冒个语法错误，但树正常构建，json也一致，但我检查找不到出错的位置，感觉没有问题，因为我们的yyerror不太容易改动，也不太熟悉Bison，现在是解决不了。

GPT: Yes, there is a way to output detailed error messages when running a binary program compiled by Bison. The Bison parser expects to report the error by calling an error reporting function named yyerror, which you must supply. It is called by yyparse whenever a syntax error is found, and it receives one argument. For a syntax error, the string is normally “syntax error”. You can implement this function to output detailed error messages that can help you debug your program. [You can also use the -v option when running Bison to generate a verbose output file that contains detailed information about the parser’s automaton computed by Bison](https://www.gnu.org/software/bison/manual/html_node/Output-Files.html)[1](https://www.gnu.org/software/bison/manual/html_node/Output-Files.html). Another option is to use the -Werror option when compiling your program with GCC. [This option treats all warnings as errors and causes GCC to stop compiling if any warnings are generated](https://stackoverflow.com/questions/62115979/how-to-implement-better-error-messages-for-flex-bison)[2](https://stackoverflow.com/questions/62115979/how-to-implement-better-error-messages-for-flex-bison). I hope this helps!

**Comment by wufeng15226** [comment]
可以自己 gdb 调试，或者给出更多信息

**Reply by QyDing02** [reply]
请问在这次作业里怎么使用gdb调试啊，有没有参考资料

**Reply by wufeng15226** [reply]
``` shell
gdb ~/sysu/bin/sysu-parser
```
https://stackoverflow.com/questions/6121094/how-do-i-run-a-program-with-commandline-arguments-using-gdb-within-a-bash-script

**Reply by QyDing02** [reply]
我把lexer的输出保存在了/tmp/temp_output.txt文件之后，在gdb里输入的指令如下图所示（即`break main`, `run /tmp/temp_output.txt`）：
![image](https://user-images.githubusercontent.com/64630300/230873834-040ffb01-9e82-4eb9-846f-a69a1c749c5e.png)
之后，gdb就无法响应我的后续输入了，按回车没反应，除非按Ctrl+C：
![image](https://user-images.githubusercontent.com/64630300/230874522-6e78ea16-cc8a-4e77-8b99-fc3db3bff097.png)

用htop查看，gdb进程的CPU占用率为0。

请问我的代码里可能出现了什么问题？

**Reply by wufeng15226** [reply]
``` shell
run < /tmp/temp_output.txt
```
